include/cuco/static_map.cuh

```diff
@@ -156,7 +153,7 @@ class static_map {
   using counter_allocator_type = typename std::allocator_traits<Allocator>::rebind_alloc<
     atomic_ctr_type>;  ///< Type of the allocator to (de)allocate atomic counters
 
-#if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 700)
+#if !defined(CUCO_HAS_INDEPENDENT_THREADS)
```

Comment:
@sleeepyjack The problem is not only about `__CUDA_ARCH_LIST__`.

This logic has an issue, e.g. when building a tarball for sm60 and sm70, we wouldn't step in this branch if sm is 70 in the old code. Now, both sm60 and sm70 will invoke the below `static_assert`s.

Reply:
The old code never worked correctly since using the `__CUDA_ARCH__` macro in host code is UB. The host code for all values of `__CUDA_ARCH__` needs to be the same iirc. Thus, if sm60 is present in the list of target architectures, we need to disable this feature globally. Maybe my logic is flawed here. Let's have an offline discussion.